internal/mcpproxy/crypto_test.go

```diff
@@ -6,78 +6,145 @@
 package mcpproxy
 
 import (
+	"fmt"
 	"testing"
 
 	"github.com/stretchr/testify/require"
 )
 
-func TestSessionEncryption(t *testing.T) {
-	sc := DefaultSessionCrypto("test", "")
-
-	enc, err := sc.Encrypt("plaintext")
-	require.NoError(t, err)
-
-	dec, err := sc.Decrypt(enc)
-	require.NoError(t, err)
-	require.Equal(t, "plaintext", dec)
+func testPBKDF2AesGcmSessionCrypto(seed, fallback string) SessionCrypto {
+	return &fallbackEnabledSessionCrypto{
+		primary:  NewPBKDF2AesGcmSessionCrypto(seed, 100),
+		fallback: NewPBKDF2AesGcmSessionCrypto(fallback, 100),
+	}
 }
 
-func TestEncryptionIsSalted(t *testing.T) {
-	sc := DefaultSessionCrypto("test", "")
-
-	enc1, err := sc.Encrypt("plaintext")
-	require.NoError(t, err)
-	enc2, err := sc.Encrypt("plaintext")
-	require.NoError(t, err)
-
-	require.NotEqual(t, enc1, enc2)
+func TestEncryptSession(t *testing.T) {
```

Comment:
The tests in this file haven't changed but I've refactored them a bit to make them trivial to use with additional crypto implementations, if find convenient to have other impls.